### PR TITLE
Updates for release 

### PR DESCRIPTION
**Pull request by cpanato**
- cleanup and adjust makefile
- add initial workflows and dependabot
- add config definitions for golangci-lint, goreleaser and ko
- add version command

Fixes: https://github.com/haydentherapper/timestamp-authority/issues/6

Did a rehearsal in my fork and here is the outcome:

- https://github.com/cpanato/timestamp-authority/releases/tag/v99.99.99

Images:
- `ghcr.io/cpanato/timestamp-server:v99.99.99`
- `ghcr.io/cpanato/timestamp-cli:v99.99.99`
- `ghcr.io/cpanato/fetch-tsa-certs:v99.99.99`

check signing

```
$ cosign verify ghcr.io/cpanato/timestamp-server:v99.99.99

Verification for ghcr.io/cpanato/timestamp-server:v99.99.99 --
The following checks were performed on each of these signatures:
  - The cosign claims were validated
  - Existence of the claims in the transparency log was verified offline
  - Any certificates were verified against the Fulcio roots.

[{"critical":{"identity":{"docker-reference":"ghcr.io/cpanato/timestamp-server"},"image":{"docker-manifest-digest":"sha256:fd9bf9a3bcb3cfed029b2df3689fa61eacc7203dbba288b704d73283fd6e8f25"},"type":"cosign container image signature"},"optional":{"1.3.6.1.4.1.57264.1.2":"push","1.3.6.1.4.1.57264.1.3":"75875aa8ce0a20a482579f8fb508839c6f3ef6b1","1.3.6.1.4.1.57264.1.4":"cut-release","1.3.6.1.4.1.57264.1.5":"cpanato/timestamp-authority","1.3.6.1.4.1.57264.1.6":"refs/tags/v99.99.99","Bundle":{"SignedEntryTimestamp":"MEUCIQCDzXRNF5z/RhEovfVtV/Y2NUcLGwh7RfFyLY9lBGEB7AIgMGVCFNBgtOGWExNkcNdWJ7Jcc4cwCrhcpLOOttp43bY=","Payload":{"body":"eyJhcGlWZXJzaW9uIjoiMC4wLjEiLCJraW5kIjoiaGFzaGVkcmVrb3JkIiwic3BlYyI6eyJkYXRhIjp7Imhhc2giOnsiYWxnb3JpdGhtIjoic2hhMjU2IiwidmFsdWUiOiJkZDI1YTQ3YTczYWFjNWRiMzQxNDM5NjI0NjM5Yzg1NWNjN2U2NmRlZjk4N2I3Y2JjZThmNGZiYzE0ZWE5YmVjIn19LCJzaWduYXR1cmUiOnsiY29udGVudCI6Ik1FWUNJUUMzVHhaWTRSWG9FNHRLWktoZ3kyUlhmZG9yVEtjWXdHeEhZUmZJb1U2dU5BSWhBS2l6NmRta2hlMzFhZnZaZUZLTkZJa1FjVXJrclN1MXlUZWhhT1lIK0h2ViIsInB1YmxpY0tleSI6eyJjb250ZW50IjoiTFMwdExTMUNSVWRKVGlCRFJWSlVTVVpKUTBGVVJTMHRMUzB0Q2sxSlNVUjBSRU5EUVhwdFowRjNTVUpCWjBsVlZFZFBNMGhyUTNadGFXOUhXVEZHUVVRNFVXazFjMDVSVldsQmQwTm5XVWxMYjFwSmVtb3dSVUYzVFhjS1RucEZWazFDVFVkQk1WVkZRMmhOVFdNeWJHNWpNMUoyWTIxVmRWcEhWakpOVWpSM1NFRlpSRlpSVVVSRmVGWjZZVmRrZW1SSE9YbGFVekZ3WW01U2JBcGpiVEZzV2tkc2FHUkhWWGRJYUdOT1RXcEpkMDlFUlRSTlZFVjZUbXBGTVZkb1kwNU5ha2wzVDBSRk5FMVVSVEJPYWtVeFYycEJRVTFHYTNkRmQxbElDa3R2V2tsNmFqQkRRVkZaU1V0dldrbDZhakJFUVZGalJGRm5RVVZyVURaSFJrcDRWakYyWkVaV1YybDBieXRhVldkMFlrTjZUaTlLV0VSdU5rNVFTMmNLWTJZelJWUXpMMmw0WlZac2RXWXJTM2xYVkROdlVubDFRMjA1VFVwalpYSllVR05MU0RNNFpsRkNiak5tUlN0clpYRlBRMEZzWjNkblowcFZUVUUwUndwQk1WVmtSSGRGUWk5M1VVVkJkMGxJWjBSQlZFSm5UbFpJVTFWRlJFUkJTMEpuWjNKQ1owVkdRbEZqUkVGNlFXUkNaMDVXU0ZFMFJVWm5VVlZ2YW5GbkNuVmthQ3RPV25KV1VVbzFOSGxYVFdoRWFXdDViR0pWZDBoM1dVUldVakJxUWtKbmQwWnZRVlV6T1ZCd2VqRlphMFZhWWpWeFRtcHdTMFpYYVhocE5Ga0tXa1E0ZDJKM1dVUldVakJTUVZGSUwwSkhWWGRaTkZwb1lVaFNNR05JVFRaTWVUbHVZVmhTYjJSWFNYVlpNamwwVERKT2QxbFhOV2hrUnpoMlpFZHNkQXBhV0U0d1dWY3hkMHhYUmpGa1IyaDJZMjFzTUdWVE9IVmFNbXd3WVVoV2FVd3paSFpqYlhSdFlrYzVNMk41T1hsYVYzaHNXVmhPYkV4dWJHaGlWM2hCQ21OdFZtMWplVGt3V1Zka2Vrd3pXVFZQVXpRMVQxTTBOVTlVUVRWQ1oyOXlRbWRGUlVGWlR5OU5RVVZDUWtOMGIyUklVbmRqZW05MlRETlNkbUV5Vm5VS1RHMUdhbVJIYkhaaWJrMTFXakpzTUdGSVZtbGtXRTVzWTIxT2RtSnVVbXhpYmxGMVdUSTVkRTFDU1VkRGFYTkhRVkZSUW1jM09IZEJVVWxGUWtoQ01RcGpNbWQzVG1kWlMwdDNXVUpDUVVkRWRucEJRa0YzVVc5T2VsVTBUbnBXYUZsVWFHcGFWRUpvVFdwQ2FFNUVaM2xPVkdNMVdtcG9iVmxxVlhkUFJHZDZDazlYVFRKYWFrNXNXbXBhYVUxVVFWcENaMjl5UW1kRlJVRlpUeTlOUVVWRlFrRjBhbVJZVVhSamJWWnpXbGRHZWxwVVFYQkNaMjl5UW1kRlJVRlpUeThLVFVGRlJrSkNkR3BqUjBaMVdWaFNka3d6VW5CaVYxWjZaRWRHZEdORE1XaGtXRkp2WWpOS2NHUklhM2RKVVZsTFMzZFpRa0pCUjBSMmVrRkNRbWRSVkFwamJWWnRZM2s1TUZsWFpIcE1NMWsxVDFNME5VOVRORFZQVkVOQ2FYZFpTMHQzV1VKQ1FVaFhaVkZKUlVGblVqbENTSE5CWlZGQ00wRkJhR2RyZGtGdkNsVjJPVzlTWkVoU1lYbGxSVzVGVm01SFMzZFhVR05OTkRCdE0yMTJRMGxIVG0wNWVVRkJRVUpuY2tNNVYzWnJRVUZCVVVSQlJXZDNVbWRKYUVGTk9UVUtMMVJIV25wcWNreGhXVEpQYVZZM2FrcHZORkI2VWtsUFltRjNaVUU0YWs1M1ZEQTFSVlZMVUVGcFJVRTRhMlJIWTA1RU4xUkNhek5YUmxBMFZpOWhid3BuUWxCdmJHOXdSREJYWjI5SmVEbHJha1U1TURGclFYZERaMWxKUzI5YVNYcHFNRVZCZDAxRVlWRkJkMXBuU1hoQlVFOWljWEpxUkRGYVVHSXhaWE5tQ2pONldIUkZRVTFKS3psVE0waEJTVU5YT0ZWMVJEZEJSSFpsVkdKYUwyTTNSblZKWW1oYVNtUm1VVFpXVmt3eVQxUlJTWGhCU1VaRFVXVk5lR0owWTBNS1lsQnVRVXRZTm5SRFRtczViMHhJZW14Q1RXeHZNMDVzTVc0MU1HeFFkSEJXVkZBdlRtcFFhbEU0Vm5JMFNreDNkamt5UWxsQlBUMEtMUzB0TFMxRlRrUWdRMFZTVkVsR1NVTkJWRVV0TFMwdExRbz0ifX19fQ==","integratedTime":1660822576,"logIndex":3213069,"logID":"c0d23d6ad406973f9559f3ba2d1ca01f84147d8ffc5b8445c224f98b9591801d"}},"GIT_HASH":"75875aa8ce0a20a482579f8fb508839c6f3ef6b1","GIT_VERSION":"v99.99.99","Issuer":"https://token.actions.githubusercontent.com","Subject":"https://github.com/cpanato/timestamp-authority/.github/workflows/release.yaml@refs/tags/v99.99.99"}}]
```

### CODE REVIEW

File: .ko.yaml
Comment (haydentherapper) [inline .ko.yaml:1]: Just wondering cause I'm not familiar, is `.ko` just used when running the `ko` commands from the Makefile locally? Or used elsewhere?

File: .goreleaser.yml
Comment (haydentherapper) [inline .goreleaser.yml:78]: We don't need to create a separate binary for this, this will only be used by production maintainers to fetch a cert once or on expiration. Running this with `go run` should be fine.

File: release/cloudbuild.yaml
Comment (haydentherapper) [inline release/cloudbuild.yaml:17]: If we remove this, will we just be using github actions to release? According to https://github.com/sigstore/public-good-instance/blob/main/playbooks/release.md, don't we need cloudbuild to push to the registries? Or should this be added only once we transfer over to sigstore (and then I'll create a ticket to track)?

**Review by haydentherapper — COMMENTED** [review]
Thank you so much for helping with this!

File: .goreleaser.yml
Comment (cpanato) [reply .goreleaser.yml:37]: we can remove

File: .goreleaser.yml
Comment (cpanato) [reply .goreleaser.yml:36]: arm is for the rasp-pies :) we can drop

File: .ko.yaml
Comment (cpanato) [reply .ko.yaml:1]: that is used when `ko` run , then it reads the `.ko.yaml` config and use that to build the image

File: .goreleaser.yml
Comment (cpanato) [reply .goreleaser.yml:78]: ok, i was unsure of that, will remove

File: release/cloudbuild.yaml
Comment (cpanato) [reply release/cloudbuild.yaml:17]: i would like to build in github actions and not use cloudbuild if dont needed, this for me sounds the case as well, it is a small and quick release.

we are doing this for policy-controller already and we use the workload identity to connect to the gcp project to push the image to there.

when we transfer this project, I will update the job to include that, for now only github registry :)

Comment (haydentherapper) [reply release/cloudbuild.yaml:17]: Sounds perfect, thanks!

**Review by haydentherapper — APPROVED** [review]
Thank you again!

**Comment by cpanato** [comment]
let me know when you are planning to transfer then i can help in the coordination and final touches

**Comment by haydentherapper** [comment]
Thanks! I opened a ticket in the TSC repo, just waiting on someone to approve.

**Pull request merged**
